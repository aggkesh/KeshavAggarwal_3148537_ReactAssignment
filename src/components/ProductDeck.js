import React, { Component } from "react";
import { CardDeck } from "react-bootstrap";
import Product from "./Product";

class ProductDeck extends Component {

    render() {
        const { products } = this.props

        const productList = [...products]
        const productCardDeckItem = []
        
        for(var index = 0; productList.length > 0; index++) {
            productCardDeckItem.push(
                <CardDeck key={ index } className="m-3" style={ 
                        productList.length === 1 ? { width: "308px" } :
                        productList.length === 2 ? { width: "616px" } : 
                        productList.length === 3 ? {width: "924px" } : null }>
                    { productList.splice(0, 4).map(product => <Product key={ product.id } product={ product } />) }
                </CardDeck>
            )
        }

        return (
            <React.Fragment>
                { productCardDeckItem }
            </React.Fragment>
        )
    }

}

export default ProductDeck