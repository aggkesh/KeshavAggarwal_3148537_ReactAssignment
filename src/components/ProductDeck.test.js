import React from 'react';
import ProductDeck from './ProductDeck';

import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

describe('Product Deck Snapshot Test Cases', () => {
    it('renders correctly enzyme', () => {
        const products = [
            {
              id: 1,
              title: "Google Pixel - Black",
              img: "/img/product-1.png",
              price: 19999,
              info: "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify."
            },
            {
              id: 2,
              title: "Samsung S7",
              img: "/img/product-2.png",
              price: 16,
              info: "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify."
            },
        ]
        const wrapper = shallow(<ProductDeck products={ products } />)
      
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});