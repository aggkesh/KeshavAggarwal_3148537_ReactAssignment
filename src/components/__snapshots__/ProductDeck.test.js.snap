// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Deck Snapshot Test Cases renders correctly enzyme 1`] = `
<Fragment>
  <CardDeck
    className="m-3"
    key="0"
    style={
      Object {
        "width": "616px",
      }
    }
  >
    <Connect(Product)
      key="1"
      product={
        Object {
          "id": 1,
          "img": "/img/product-1.png",
          "info": "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify.",
          "price": 19999,
          "title": "Google Pixel - Black",
        }
      }
    />
    <Connect(Product)
      key="2"
      product={
        Object {
          "id": 2,
          "img": "/img/product-2.png",
          "info": "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify.",
          "price": 16,
          "title": "Samsung S7",
        }
      }
    />
  </CardDeck>
</Fragment>
`;
